signac_dashboard/dashboard.py

```diff
@@ -616,3 +633,6 @@ def _run(args):
             sys.exit(1)
         else:
             sys.exit(0)
+        finally:
+            self.observer.stop()
+            self.observer.join()
```

Comment:
Does it make sense to have a `finally` clause after `sys.exit()` has been called?

Reply:
I agree that it looks a little weird, and the `exit()` call is executed prior to the finally block, but the finally block is definitely executed regardless.

I actually don't think that there is a reason to explicitly call `sys.exit()` here, so one solution might be to just remove those calls. Is exiting the application gracefully via interrupt considered an error state?

Reply:
This explains why the `finally` is still executed: https://stackoverflow.com/questions/7709411/why-finally-block-is-executing-after-calling-sys-exit0-in-except-block

Let's remove the `else: sys.exit(0)` clause.

Based on this page, I think the keyboard interrupt should continue to exit with a nonzero status: https://unix.stackexchange.com/questions/251996/why-does-bash-set-exit-status-to-non-zero-on-ctrl-c-or-ctrl-z

Reply:
Done.

Reply:
As far as I can tell, the `KeyboardInterrupt` exception handler is never executed, because flask is capturing this. The exit code is 0.